### PR TITLE
Fix code style and add test case

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/chaingovernance/chain_governance.go  2
- precompiled/chaingovernance/chain_governance_test.go  5
- precompiled/chaingovernance/chain_governance_service.go  4
- precompiled/chaingovernance/account_info.go  1
         

Clones added
============
- precompiled/chaingovernance/chain_governance.go  5
- precompiled/chaingovernance/chain_governance_test.go  2
- precompiled/chaingovernance/chain_governance_service.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/chaingovernance/chain_governance_service.go  4
- precompiled/chaingovernance/chain_governance_test.go  5
- precompiled/chaingovernance/account_info.go  1
- precompiled/chaingovernance/chain_governance.go  2
         

Clones added
============
- precompiled/chaingovernance/chain_governance_service.go  1
- precompiled/chaingovernance/chain_governance_test.go  3
- precompiled/chaingovernance/chain_governance.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/chaingovernance/account_info.go  1
- precompiled/chaingovernance/chain_governance_service.go  4
- precompiled/chaingovernance/chain_governance.go  2
- precompiled/chaingovernance/chain_governance_test.go  5
         

Clones added
============
- precompiled/chaingovernance/chain_governance_service.go  1
- precompiled/chaingovernance/chain_governance.go  5
- precompiled/chaingovernance/chain_governance_test.go  2
- precompiled/permission/permission_service.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/chaingovernance/chain_governance.go  2
- precompiled/chaingovernance/chain_governance_test.go  5
- precompiled/chaingovernance/chain_governance_service.go  4
- precompiled/chaingovernance/account_info.go  1
         

Clones added
============
- precompiled/consensus/consensus_service_test.go  2
- precompiled/permission/permission_service.go  1
- precompiled/permission/permission_service_test.go  2
- precompiled/chaingovernance/chain_governance.go  5
- precompiled/chaingovernance/chain_governance_test.go  4
- precompiled/chaingovernance/chain_governance_service.go  1
- precompiled/cns/cns_service_test.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

File: precompiled/chaingovernance/account_info.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package chaingovernance
+
+type AccountInfo struct {
+	Address   string `json:"address"`
+	EnableNum string `json:"enable_num"`
+}
+
+func (c *AccountInfo) GetAddress() string {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/account_info.go:8]: delete these

File: precompiled/chaingovernance/chain_governance_service.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	chainGovernance     *ChainGovernance
+	chainGovernanceAuth *bind.TransactOpts
+	client              *client.Client
+}
+
+// chainGovernancePrecompileAddress is the contract address of Permission
+var chainGovernancePrecompileAddress common.Address = common.HexToAddress("0x0000000000000000000000000000000000001008")
+
+// NewChainGovernanceService returns ptr of ChainGovernanceService
+func NewChainGovernanceService(client *client.Client, privateKey *ecdsa.PrivateKey) (*ChainGovernanceService, error) {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:28]:
```suggestion
func NewChainGovernanceService(client *client.Client) (*ChainGovernanceService, error) {
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+	instance, err := NewChainGovernance(chainGovernancePrecompileAddress, client)
+	if err != nil {
+		return nil, fmt.Errorf("construct ChainGovernanceService failed: %+v", err)
+	}
+	auth := bind.NewKeyedTransactor(privateKey)
+	auth.GasLimit = big.NewInt(30000000)
+	return &ChainGovernanceService{instance, auth, client}, nil
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:35]:
```suggestion
	return &ChainGovernanceService{instance, client.GetTransactOpts(), client}, nil
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+// GrantCommitteeMember grants committee member
+func (service *ChainGovernanceService) GrantCommitteeMember(accountAddress string) (string, error) {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:39]:
```suggestion
func (service *ChainGovernanceService) GrantCommitteeMember(accountAddress common.Address) (string, error) {
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	tx, err := service.chainGovernance.GrantCommitteeMember(service.chainGovernanceAuth, common.HexToAddress(accountAddress))
+	if err != nil {
+		return "", fmt.Errorf("ChainGovernanceService GrantCommitteeMember failed: %v", err)
+	}
+	return handleReceipt(service.client, tx)
+}
+
+// RevokeCommitteeMember revokes committee member
+func (service *ChainGovernanceService) RevokeCommitteeMember(accountAddress string) (string, error) {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:48]:
```suggestion
func (service *ChainGovernanceService) RevokeCommitteeMember(accountAddress string) (int, error) {
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+	// wait for the mining
+	receipt, err := bind.WaitMined(context.Background(), c, tx)
+	if err != nil {
+		return "", fmt.Errorf("ChainGovernanceService wait for the transaction receipt failed: %v", err)
+	}
+	status := receipt.GetStatus()
+	if types.Success != status {
+		return "", fmt.Errorf(types.GetStatusMessage(status))
+	}
+	output := receipt.GetOutput()
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:191]: abi parse output

File: precompiled/chaingovernance/chain_governance_service.go [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+		return "", fmt.Errorf("ChainGovernanceService UpdateCommitteeMemberWeight failed: %v", err)
+	}
+	return handleReceipt(service.client, tx)
+}
+
+// UpdateThreshold updates the threshold that the committee vote needs to reach
+func (service *ChainGovernanceService) UpdateThreshold(threshold *big.Int) (string, error) {
+	upperBound := threshold.Cmp(big.NewInt(100))
+	lowerBound := threshold.Cmp(big.NewInt(0))
+	if upperBound == 1 || lowerBound == -1 {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:100]:
```suggestion
	if threshold.Cmp(big.NewInt(100)) || big.NewInt(0).Cmp(threshold) {
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+	tx, err := service.chainGovernance.UpdateCommitteeMemberWeight(service.chainGovernanceAuth, common.HexToAddress(accountAddress), weight)
+	if err != nil {
+		return "", fmt.Errorf("ChainGovernanceService UpdateCommitteeMemberWeight failed: %v", err)
+	}
+	return handleReceipt(service.client, tx)
+}
+
+// UpdateThreshold updates the threshold that the committee vote needs to reach
+func (service *ChainGovernanceService) UpdateThreshold(threshold *big.Int) (string, error) {
+	upperBound := threshold.Cmp(big.NewInt(100))
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:98]:
```suggestion
	upperBound := threshold.Cmp(big.NewInt(99))
```

File: precompiled/chaingovernance/chain_governance_service.go [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+	}
+	tx, err := service.chainGovernance.UpdateCommitteeMemberWeight(service.chainGovernanceAuth, common.HexToAddress(accountAddress), weight)
+	if err != nil {
+		return "", fmt.Errorf("ChainGovernanceService UpdateCommitteeMemberWeight failed: %v", err)
+	}
+	return handleReceipt(service.client, tx)
+}
+
+// UpdateThreshold updates the threshold that the committee vote needs to reach
+func (service *ChainGovernanceService) UpdateThreshold(threshold *big.Int) (string, error) {
```
Comment (bxq2011hust) [inline precompiled/chaingovernance/chain_governance_service.go:97]:
```suggestion
func (service *ChainGovernanceService) UpdateThreshold(threshold *big.Int) (*big.Int, error) {
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/goclient.go  1
- conf/config_test.go  3
- conf/parsepem_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/goclient.go  1
- conf/parsepem_test.go  5
- conf/config_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/api_handler.go  6
- conf/parse_pem_test.go  5
- precompiled/cns/cns_info.go  2
- conf/parse_pem.go  9
- client/go_client.go  11
- conf/config_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- .ci/ethPrecompiled/bn256.go  1
- conf/parse_pem.go  9
- client/go_client.go  11
- precompiled/cns/cns_info.go  2
- client/api_handler.go  6
- conf/parse_pem_test.go  5
- .ci/ethPrecompiled/bn256_gm.go  1
- conf/config_test.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/cns/cns_info.go  2
- client/api_handler.go  6
- .ci/ethPrecompiled/bn256_gm.go  1
- client/go_client.go  11
- conf/parse_pem.go  9
- conf/parse_pem_test.go  5
- .ci/ethPrecompiled/bn256.go  1
- conf/config_test.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- conf/parse_pem_test.go  5
- conf/config_test.go  3
- client/api_handler.go  6
- precompiled/cns/cns_info.go  2
- client/go_client.go  11
- conf/parse_pem.go  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5853660&bid=19163508)

**Pull request closed without merging**